src/app/fdctl/run/tiles/fd_resolv.c

```diff
@@ -119,11 +163,33 @@ after_frag( fd_resolv_ctx_t *   ctx,
   (void)chunk;
 
   if( FD_UNLIKELY( ctx->in[in_idx].kind==FD_RESOLV_IN_KIND_BANK ) ) {
-    fd_rooted_bank_t * frag = (fd_rooted_bank_t *)ctx->_bank_msg;
-    if( FD_LIKELY( ctx->root_bank ) ) fd_ext_bank_release( ctx->root_bank );
-
-    ctx->root_bank = frag->bank;
-    ctx->root_slot = frag->slot;
+    switch( sig ) {
+      case 0: {
+        fd_rooted_bank_t * frag = (fd_rooted_bank_t *)ctx->_bank_msg;
+        if( FD_LIKELY( ctx->root_bank ) ) fd_ext_bank_release( ctx->root_bank );
+
+        ctx->root_bank = frag->bank;
+        ctx->root_slot = frag->slot;
+        break;
+      }
+      case 1: {
+        fd_completed_bank_t * frag = (fd_completed_bank_t *)ctx->_bank_msg;
+
+        blockhash_map_t * entry = map_query( ctx->blockhash_map, ctx->blockhash_ring[ ctx->blockhash_ring_idx%4096UL ], NULL );
+        if( FD_LIKELY( entry ) ) map_remove( ctx->blockhash_map, entry );
+
+        fd_memcpy( ctx->blockhash_ring[ ctx->blockhash_ring_idx%4096UL ].b, frag->hash, 32UL );
```

Comment:
Normal memcpy is probably better here, since size is small and known at compile time. The compiler normally has an easier time optimizing it.